### ISSUE TITLE
Make SelectField.choices accept a callable

### ISSUE THREAD

**Issue by pbu88** [wtforms/wtforms#185]
Hi,

Would it be so hard to make `choices` argument accept a callable too? I didn't go deep by looking at the code, but you currently have this:

```
class SelectField(SelectFieldBase):
    widget = widgets.Select()

    def __init__(self, label=None, validators=None, coerce=text_type, choices=None, **kwargs):
        super(SelectField, self).__init__(label, validators, **kwargs)
        self.coerce = coerce
        self.choices = choices
```

I wonder if we can just modify last line to something like:

```
if callable(choices):
    # make it callable with no arguments to keep it at its simplest.
    self.choices = choices()
else:
    self.choices = choices
```

This looks very simple and would make the field more powerful. For example, there are things that when the module loads, are not quite ready (I'm thinking about flask context for example, but can be anything actually) but when the Form is actually instantiated, they are ready. 

I'm not sure if this has been discussed before and not very sure about the implications, but I think can be a good yet simple enhancement. I would be gladly to do it myself and test it if you need the help.

**Comment by jmagnusson** [comment]
In case you're looking for a solition now I can inform you that [wtforms-components](https://wtforms-components.readthedocs.org/en/latest/) already supports this in its subclassed Select(Multiple)Field.

**Comment by roganov** [comment]
I you don't have the choices on import time, then IMO you be passing the choices to the form constructor. This will simplify unit-testing and the users (other programmers on the team, even yourself) of the form will be aware of the fact that your form requires some external resource.

**Comment by ftm** [comment]
As given in the documentation, the recommended way of doing this is to not pass a `choices` when declaring the `SelectField` parameter but instead create the list of choices when ready and assign them.

For example:
```python
class UserDetails(Form):
    group_id = SelectField(u'Group', coerce=int)

def edit_user(request, id):
    user = User.query.get(id)
    form = UserDetails(request.POST, obj=user)
    form.group_id.choices = [(g.id, g.name) for g in Group.query.order_by('name')]
```

**Comment by azmeuk** [comment]
Fixed by #608
